eclipse/src/saros/ui/wizards/pages/LocalRepresentationSelectionPage.java

```diff
@@ -219,6 +220,18 @@ private void showNewTabErrorMessage(SelectionEvent e) {
           }
         });
 
+    Button newProjectButton = new Button(composite, SWT.PUSH);
+    newProjectButton.setImage(ImageManager.ETOOL_NEW_PROJECT);
+    newProjectButton.setLayoutData(new GridData(SWT.END, SWT.FILL, false, false));
+    newProjectButton.setText(Messages.LocalRepresentationSelectionPage_new_project_button);
+    newProjectButton.addSelectionListener(
+        new SelectionAdapter() {
+          @Override
+          public void widgetSelected(SelectionEvent e) {
+            WizardUtils.openNewProjectWizard();
```

Comment:
Can you apply a TODO to correctly configure the project (project nature etc., of course nothing of the required data is currently transmitted) ?

Reply:
It is not that easy to change the project nature afterwards. Most of the times you have to convert the project to a facetted project which is not the same as, e.g a pure Java project.

Reply:
> It is not that easy to change the project nature afterwards. Most of the times you have to convert the project to a facetted project which is not the same as, e.g a pure Java project.

I am not that familiar with Eclipse project natures, so I can't speak to that. But I would assume that the default java project configuration (or any other configuration that can be created through the normal Eclipse project creation wizard) is usable in most cases (even if it reduces the available feature set).

Either way, this is not the place to put a TODO for this in my opinion. When creating the project, we have no idea which shared reference point it will actually be used for. So the logic would have to be added to the resource negotiation. The only option to even potentially offer correctly setting up the project is to send the necessary data as additional resource negotiation data in my opinion. With any other approach, we would be adding IDE-specific logic/concepts to core logic.
(Additionally, changing the project configuration might be confusing to the user in cases where only inner resources (i.e. not a project root) are shared. In such a setup, adjusting the project configuration would modify resources that are not shared.)

But I don't think that this should be part of this patch. If you are interested in pursuing such an approach (or want to create a reminder for a later date), I would suggest opening a new issue for it.